File: src/jk_5/nailed/groups/GroupRegistry.java
package jk_5.nailed.groups;

import com.google.common.collect.BiMap;
import com.google.common.collect.HashBiMap;
import com.google.common.collect.Maps;

import java.util.Map;

/**
 * TODO: Edit description
 *
 * @author jk-5
 */
public class GroupRegistry {

    final BiMap<String, Group> groups = HashBiMap.create();
    private Group defaultGroup;

    public void registerGroup(String name, Group group){
        this.groups.put(name, group);
    }

    public Group getGroup(String name){
        return this.groups.get(name);
    }

    public Group getDefaultGroup(){
        return this.defaultGroup;
    }

    public void setDefaultGroup(String name){
        Group g = this.getGroup(name);
        if(g != null) this.defaultGroup = g;
    }

    public Iterable<String> getGroups(){
        return this.groups.keySet();
    }
}

File: src/jk_5/nailed/groups/GroupRegistry.scala
package jk_5.nailed.groups

import scala.collection.mutable

/**
 * No description given
 *
 * @author jk-5
 */
object GroupRegistry {

  private final val groups = mutable.HashMap[String, Group]()
  private var defaultGroup: Group = _

  @inline def registerGroup(name: String, group: Group) = this.groups.put(name, group)
  @inline def getGroup(name: String) = this.groups.get(name)
  @inline def getGroupID(group: Group) = this.groups.find(_._2 == group).get._1
  @inline def getDefaultGroup = this.defaultGroup
  @inline def setDefaultGroup(name: String) = this.getGroup(name).foreach(g => this.defaultGroup = g)
  @inline def getGroups = this.groups.keySet
}
